cmake/FindFreeRTOSPlus.cmake

```diff
@@ -0,0 +1,149 @@
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/psa_crypto.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/psa_crypto_driver_wrappers.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/psa_crypto_se.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/psa_crypto_slot_management.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/psa_crypto_storage.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/psa_its_file.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ripemd160.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/rsa.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/rsa_internal.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/sha1.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/sha256.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/sha512.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_cache.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_ciphersuites.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_cli.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_cookie.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_msg.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_srv.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_ticket.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_tls.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/ssl_tls13_keys.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/threading.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/timing.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/version.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/version_features.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509_create.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509_crl.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509_crt.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509_csr.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509write_crt.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/x509write_csr.c
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/library/xtea.c
+        ${FreeRTOSPlus_PATH}//Source/Utilities/mbedtls_freertos/mbedtls_error.c)
+    target_include_directories(FreeRTOSPlus::ThirdParty::mbedtls INTERFACE
+        ${FreeRTOSPlus_PATH}/ThirdParty/mbedtls/include)
+    target_include_directories(FreeRTOSPlus::ThirdParty::mbedtls INTERFACE
+        ${FreeRTOSPlus_PATH}/Source/Utilities/mbedtls_freertos)
+    target_compile_definitions(FreeRTOSPlus::ThirdParty::mbedtls INTERFACE
+        MBEDTLS_CONFIG_FILE="mbedtls_config.h")
+endif()
+
+if(NOT (TARGET FreeRTOSPlus::TCPIP))
+    add_library(FreeRTOSPlus::TCPIP INTERFACE IMPORTED)
+    target_sources(FreeRTOSPlus::TCPIP INTERFACE
```

Comment:
does everything here need to be INTERFACE? INTERFACE is only public facing (not PRIVATE). If you make it PUBLIC it's in INTERFACE and PRIVATE

Reply:
discuss offline... for now I will keep it same